seed-job/index_reader.py

```diff
@@ -0,0 +1,352 @@
+"""
+This script parses the container index specified and
+creates the Jenkins pipeline projects from entries of index.
+"""
+
+import subprocess
+import sys
+import yaml
+
+from glob import glob
+
+
+def run_cmd(cmd, shell=False):
+    """
+    Runs the given shell command
+
+    :param cmd: Command to run
+    :param shell: Whether to run raw shell commands with '|' and redirections
+    :type cmd: str
+    :type shell: boolean
+
+    :return: Command output
+    :rtype: str
+    :raises: subprocess.CalledProcessError
+    """
+    if shell:
+        return subprocess.check_output(cmd, shell=True)
+    else:
+        return subprocess.check_output(cmd.split(), shell=False)
+
+
+class Project(object):
+    """
+    Class for storing and processing a container index project
+    """
+
+    def __init__(self, entry, namespace):
+        """
+        Initialize project object with an entry in container index
+        """
+        self.namespace = namespace
+        self.load_project_entry(entry)
+        self.pipeline_name = self.get_pipeline_name()
+
+    def __str__(self):
+        """
+        Returns the string representation of the project object
+        It returns the pipeline-name, which is constructed
+        based on parameters of the project indexed.
+        """
+        return self.pipeline_name
+
+    def replace_dot_slash_colon_(self, value):
+        """
+        Given a value with either dot slash or underscore,
+        replace each with hyphen
+        """
+        return value.replace("_", "-").replace("/", "-").replace(
+            ".", "-").replace(":", "-")
+
+    def process_depends_on(self, depends_on=None):
+        """
+        Process depends_on for given project based on entry index
+        and namespace
+        """
+        if not depends_on or depends_on == "null":
+            return None
+
+        if isinstance(depends_on, list):
+            return ",".join("{}-{}".format(
+                self.namespace,
+                self.replace_dot_slash_colon_(d))
+                for d in depends_on)
+        else:
+            return "{}-{}".format(
+                self.namespace,
+                self.replace_dot_slash_colon_(depends_on))
+
+    def process_desired_tag(self, desired_tag=None):
+        """
+        Process desired_tag for given project
+        """
+        if not desired_tag:
+            return "latest"
+        return desired_tag
+
+    def process_pre_build_script(self, prebuild_script=None):
+        """
+        Process prebuild_script for given project
+        """
+        if not prebuild_script:
+            return None
+        return prebuild_script
+
+    def process_pre_build_context(self, prebuild_context=None):
+        """
+        Process prebuild_context for given project
+        """
+        if not prebuild_context:
+            return None
+        return prebuild_context
```

Comment:
This and above methods can be eliminated by inline processing in load_projects method,
however I kept it so that if needed we could perform some steps in prebuild scenario. We can remove them as well. @bamachrn : WDYT for prebuild scenario?